libs/types/src/fixed_point.rs

```diff
@@ -85,1511 +85,1673 @@ fn to_bound<N: FixedPointOperand, D: FixedPointOperand, R: Bounded>(n: N, m: D)
-					((self.0 % Self::accuracy()) as i128).abs(),
-					weight = precision
-				);
-				write!(f, "{}({}.{})", stringify!($name), integral, fractional)
-			}
+	fn neg(self) -> Self::Output {
+		Self(<Self as FixedPointNumber>::Inner::zero() - self.0)
+	}
+}
 
-			#[cfg(not(feature = "std"))]
-			fn fmt(&self, _: &mut sp_std::fmt::Formatter) -> sp_std::fmt::Result {
-				Ok(())
-			}
-		}
+impl ops::Add for Rate {
+	type Output = Self;
 
-		#[cfg(feature = "std")]
-		impl sp_std::fmt::Display for $name {
-			fn fmt(&self, f: &mut sp_std::fmt::Formatter) -> sp_std::fmt::Result {
-				write!(f, "{}", self.0)
-			}
-		}
+	fn add(self, rhs: Self) -> Self::Output {
+		Self(self.0 + rhs.0)
+	}
+}
 
-		#[cfg(feature = "std")]
-		impl sp_std::str::FromStr for $name {
-			type Err = &'static str;
+impl ops::Sub for Rate {
+	type Output = Self;
 
-			fn from_str(s: &str) -> Result<Self, Self::Err> {
-				let inner: <Self as FixedPointNumber>::Inner = s
-					.parse()
-					.map_err(|_| "invalid string input for fixed point number")?;
-				Ok(Self::from_inner(inner))
-			}
-		}
+	fn sub(self, rhs: Self) -> Self::Output {
+		Self(self.0 - rhs.0)
+	}
+}
 
-		// Manual impl `Serialize` as serde_json does not support i128.
-		// TODO: remove impl if issue https://github.com/serde-rs/json/issues/548 fixed.
-		#[cfg(feature = "std")]
-		impl Serialize for $name {
-			fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-			where
-				S: Serializer,
-			{
-				serializer.serialize_str(&self.to_string())
-			}
-		}
+impl ops::Mul for Rate {
+	type Output = Self;
 
-		// Manual impl `Deserialize` as serde_json does not support i128.
-		// TODO: remove impl if issue https://github.com/serde-rs/json/issues/548 fixed.
-		#[cfg(feature = "std")]
-		impl<'de> Deserialize<'de> for $name {
-			fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
-			where
-				D: Deserializer<'de>,
-			{
-				use sp_std::str::FromStr;
-				let s = String::deserialize(deserializer)?;
-				$name::from_str(&s).map_err(de::Error::custom)
-			}
+	fn mul(self, rhs: Self) -> Self::Output {
+		self.checked_mul(&rhs)
+			.unwrap_or_else(|| panic!("attempt to multiply with overflow"))
+	}
+}
+
+impl ops::Div for Rate {
+	type Output = Self;
+
+	fn div(self, rhs: Self) -> Self::Output {
+		if rhs.0 == 0 {
+			panic!("attempt to divide by zero")
 		}
+		self.checked_div(&rhs)
+			.unwrap_or_else(|| panic!("attempt to divide with overflow"))
+	}
+}
 
-		#[cfg(test)]
-		mod $test_mod {
-			use super::*;
+impl CheckedSub for Rate {
+	fn checked_sub(&self, rhs: &Self) -> Option<Self> {
+		self.0.checked_sub(rhs.0).map(Self)
+	}
+}
 
-			fn max() -> $name {
-				$name::max_value()
-			}
+impl CheckedAdd for Rate {
+	fn checked_add(&self, rhs: &Self) -> Option<Self> {
+		self.0.checked_add(rhs.0).map(Self)
+	}
+}
 
-			fn min() -> $name {
-				$name::min_value()
-			}
+impl CheckedDiv for Rate {
+	fn checked_div(&self, other: &Self) -> Option<Self> {
+		self.checked_div_floor(other)
```

Comment:
Same as above. If not explicitly used, I would like to have the current `NearestPrefLow` as default.